### PR TITLE
clean poster elements at the end of its flush operation

### PR DESCRIPTION
**Pull request by yahonggu**

### CODE REVIEW

File: libs/poster.py [hunk 1]
```diff
@@ -180,6 +180,7 @@ def flush(self):
                 self.packet_count = 0
                 self.event_count = 0
                 self.events = []
+                self.elements.delete_all()
```
Comment (TheConnMan) [inline libs/poster.py:183]: Is there any way for samples to come in between the time other samples are posted and these elements are cleared? In other words, will we drop samples at all by clearing the elements here?

Comment (yahonggu) [reply libs/poster.py:183]: no, we're not dropping samples, once it is flushed, and a new sample/metric comes in, the element will be recreated if not there.

Comment (TheConnMan) [reply libs/poster.py:183]: Sorry, I meant if there's a race condition. This running in the background vs incoming payloads on a different thread.

Comment (yahonggu) [reply libs/poster.py:183]: There is a "with self.lock" protecting this block on line 73 that should address your concern of race condition.

**Review by TheConnMan — APPROVED** [review]

**Comment by TheConnMan** [comment]
Can you investigate the build failures before merging?

**Comment by yahonggu** [comment]
The travis error log has null in it without any specific information on what's failed. since I have run the relevant unittest test_poster.py locally with no errors, I think this travis failure is environmental or some unrelated change before this PR.

**Comment by TheConnMan** [comment]
Can you restart the builds? Master and develop both passed the last time. If your branch fails then develop will most likely also fail when you merge. Travis runs these entirely in Docker containers.

The Travis builds test multiple versions of Python and do more than just run the tests.

**Comment by TheConnMan** [comment]
LGTM

**Comment by yahonggu** [comment]
Yeah, rerun travis works. I'll merge.

**Pull request merged**